next-frontend/src/globals/Nav.ts

```diff
@@ -1,5 +1,89 @@
 import { Block, GlobalConfig } from "payload";
 
+const iconOptions = [
```

Comment:
Ah, here's _almost_ the type defintion I asked for above.
Unfortunately, you cannot include types as dropdown choice, but if you build an `iconMap` anyways you can "hack" your way around that restriction by using `Object.keys(iconMap)` here.